flutter/documentation/adding-custom-backend.md

```diff
@@ -1,45 +1,57 @@
 
-This file describes how to add support for your backend.
+This file describes how to add support for your backend into the Flutter app.
 
 # Contents
 
+* [Folder structure](#folder-structure)
+* [Integrating your backend into the app](#integrating-your-backend-into-the-app)
 * [Implementing backend interface](#implementing-backend-interface)
-* [Adding backend to list of backends](#adding-backend-to-list-of-backends)
-* [Building the app with backends](#building-the-app-with-backends)
 
-# Implementing backend interface
-
-The following steps are required to add your backend:
+# Folder structure
 
-1. Create a mobile_back_[vendor] top level directory that builds your backend using android/cpp/c/backend_c.h
-2. Implement all the C API backend interfaces defined in android/cpp/c/backend_c.h
-3. Add a bazel BUILD file to create lib[vendor]backend.so shared library
+Create a new folder for your backend, named `mobile_back_<vendor>`, where `<vendor>` is the name of your organization.
+Structure inside that your backend directory in not strictly regulated, you may keep it the way that is convenient to you.
+See existing `mobile_back_*` folders for examples on how you you can structure your files.
 
-You can look at the [default TFLite backend implementation](../../mobile_back_tflite) for a reference.
+Generally it's a good idea to keep all C++ files in a `cpp` folder.
 
-# Adding backend to list of backends
+Add a readme file explaining how to set up environment to build your backend and which commands to use.
 
-1. Modify file [backends_list.in](../lib/backend/backends_list.in).
-Add a line with your backend tag.
-```dart
-const _backendsList = [
-    VENDOR_TAG
-    TFLITE_TAG
-];
+Add `<vendor>_backend.mk` file defining variables for your backend, similar to such files for other backends. This file should have the following structure:
+```make
+ifeq (${WITH_<VENDOR>},1)
+  $(info WITH_<VENDOR>K=1)
```

Comment:
Is the `K` just a typo, or does it have some meaning?

Reply:
typo

Reply:
fixed